ksqldb-cli/src/test/java/io/confluent/ksql/cli/BasicAuthFunctionalTest.java

```diff
@@ -165,44 +164,55 @@ private static int canMakeCliRequest(final BasicCredentials credentials) {
     )) {
       final RestResponse<ServerInfo> response = restClient.getServerInfo();
       if (response.isSuccessful()) {
-        return Code.OK.getCode();
+        return OK.code();
       }
 
       return response.getErrorMessage().getErrorCode();
     }
   }
 
-  private static Code makeWsRequest(final Optional<BasicCredentials> creds) throws Exception {
-    final WebSocketClient wsClient = new WebSocketClient();
-    wsClient.start();
+  private static int makeWsRequest(final Optional<BasicCredentials> creds) throws Exception {
 
+    Vertx vertx = Vertx.vertx();
+    HttpClient httpClient = null;
     try {
-      final ClientUpgradeRequest request = new ClientUpgradeRequest();
-      creds.ifPresent(c -> {
-        final String authHeader = "Basic " + buildBasicAuthHeader(c.username(), c.password());
-        request.setHeader(HttpHeaders.AUTHORIZATION, authHeader);
-      });
+      httpClient = vertx.createHttpClient();
 
-      final WebSocketListener listener = new WebSocketListener();
-      final URI wsUri = REST_APP.getWsListener().resolve("/ws/query");
+      final URI uri = REST_APP.getWsListener().resolve("/ws/query");
 
-      wsClient.connect(listener, wsUri, request);
+      final MultiMap headers = MultiMap.caseInsensitiveMultiMap();
 
-      assertThat("Response received",
-          listener.latch.await(30, TimeUnit.SECONDS), is(true));
```

Comment:
Is there an equivalent of this 30 second timeout we can use in the new implementation? Or is 30 seconds the default test timeout, in which case this might not matter

Reply:
ack